wayrs-client/src/lib.rs

```diff
@@ -18,8 +18,51 @@ pub use wayrs_core as core;
 pub use wayrs_core::{Fixed, IoMode};
 
 use std::ffi::CStr;
-use std::fmt;
+use std::os::unix::net::UnixStream;
+use std::path::PathBuf;
+use std::{env, fmt};
 
+pub use wayrs_core::transport::Transport;
+
+pub trait ClientTransport: Transport {
+    fn connect() -> Result<Self, ConnectError>
+    where
+        Self: Sized;
+
+    fn fix_metadata(
+        &mut self,
+        plane_idx: usize,
+        width: u32,
+        height: u32,
+        format: u32,
+    ) -> Option<(u32, u32, u64)>;
```

Comment:
- This function really needs some docs.
- This is very `wayrs-egl` specific. Maybe it should be moved to its own `trait EglTransport: ClientTransport`?

Reply:
Sounds like a good idea, I'll refactor it and add some docs soon